src/middlewared/middlewared/plugins/cluster_linux/ctdb_public.py

```diff
@@ -170,13 +171,30 @@ async def do_create(self, job, data):
 
         schema_name = 'public_create'
         verrors = ValidationErrors()
+        if 'pnn' not in data:
+            data['pnn'] = await self.middleware.call('ctdb.general.pnn')
 
         await self.middleware.call('ctdb.ips.common_validation', data, schema_name, verrors)
         await self.middleware.call('ctdb.ips.update_file', data, schema_name)
         await self.middleware.call('ctdb.public.ips.reload')
 
         return await self.middleware.call('ctdb.public.ips.query', [('id', '=', data['pnn'])])
 
+    @accepts(Dict(
+        'delip_payload',
+        Int('pnn', required=False),
+        IPAddr('public_ip', required=True)
+    ))
+    def delete_ip(self, data):
```

Comment:
We should probably protect this with a lock that is shared between `do_create` and `do_update` since it's a public endpoint, right? Or should this be private?

Reply:
Ah, forgot to make it private. The accepts() decorator is there for normalization.